easy-random-core/src/main/java/org/jeasy/random/EasyRandom.java

```diff
@@ -97,7 +97,7 @@ public EasyRandom(final EasyRandomParameters easyRandomParameters) {
      * @throws ObjectCreationException when unable to create a new instance of the given type
      */
     public <T> T nextObject(final Class<T> type) {
-        return doPopulateBean(type, new RandomizationContext(type, parameters));
+        return doPopulateBean(type, new RandomizationContext(type, parameters, randomizerProvider));
```

Comment:
No need to pass the `randomizerProvider` as a third parameter here, since it can be accessed through the parameters, which are part of the context.

Reply:
`randomizerProvider` and `parameters.getRandomizerProvider()` are different, and in canonical use, `parameters.getRandomizerProvider()` will be `null`:
https://github.com/j-easy/easy-random/blob/4b46b05a44cf1f606624a34fb783b31816107344/easy-random-core/src/main/java/org/jeasy/random/EasyRandom.java#L77

Reply:
Well this is the problem then, they should _not_ be different. You should get either the custom provider if any, or the default one, but not null. This is actually an initialization issue in ER, which was due to a wrong assertion (1912e107f1d2cce4313d3993b552f72308f970e1) and which I fixed in f0b8851f28ed88ed7131d694034a5d521d374be5. Can you please give the snapshots a try? You should now be able to get the randomizer provider from the parameters.

Reply:
Great, this will indeed help solving my problem!
I can now close this PR :)